core/camel-support/src/main/java/org/apache/camel/saga/InMemorySagaCoordinator.java

```diff
@@ -75,6 +75,13 @@ public String getId() {
 
     @Override
     public CompletableFuture<Void> beginStep(Exchange exchange, CamelSagaStep step) {
+        Status status = currentStatus.get();
+        if (status != Status.RUNNING) {
+            CompletableFuture<Void> res = new CompletableFuture<>();
+            res.completeExceptionally(new IllegalStateException("Cannot begin: status is " + status));
+            return res;
+        }
+
```

Comment:
Would you mind adding a test case somewhere in https://github.com/apache/camel/blob/main/core/camel-core/src/test/java/org/apache/camel/processor to prove that it fixes the bug?

Reply:
Sure and thanks for pointing the src directory out. I was thinking about a test case but have not find the right place.